packages/samples/headless-ssr/app-router/src/app/generic/common/engine.ts

```diff
@@ -10,4 +12,20 @@ const engineDefinition = defineSearchEngine(config);
 export type SearchStaticState = InferStaticState<typeof engineDefinition>;
 export type SearchHydratedState = InferHydratedState<typeof engineDefinition>;
 
-export const {fetchStaticState, hydrateStaticState} = engineDefinition;
+export const {
+  fetchStaticState: {fromBuildResult: fetchStaticState},
+  hydrateStaticState: {fromBuildResult: hydrateStaticState},
+} = engineDefinition;
+
+export async function fetchBuildResult(options: {
+  searchParametersInitialState: SearchParameterManagerInitialState;
+}) {
+  const buildResult = await engineDefinition.build({
+    controllers: {
+      searchParameters: {initialState: options.searchParametersInitialState},
+    },
+  });
+  const {registerNumberOfResults} = loadPaginationActions(buildResult.engine);
+  buildResult.engine.dispatch(registerNumberOfResults(6));
```

Comment:
Any reason for choosing the number 6 instead of the default 10?
Might be good to name it in a var to better document it.

Reply:
it was just temporary to illustrate in this PR what the new `.fromBuildResult` enables, I will revert all changes to samples (as mentioned at the end of the PR description).